posthog/middleware.py

```diff
@@ -188,3 +189,31 @@ def switch_team_if_needed_and_possible(self, request: HttpRequest, target_querys
                     user.save()
                     # Information for POSTHOG_APP_CONTEXT
                     setattr(request, "switched_team", current_team.id)
+
+
+from rest_framework import parsers
+
+
+class FormJSONParser(parsers.FormParser):
+    """
+    Custom FormParser that parses any JSON
+    """
+
+    def parse(self, stream, media_type=None, parser_context=None):
+        result = super().parse(stream, media_type=media_type, parser_context=parser_context)
+        data = {}
+
+        for key, value in result.items():
+            if type(value) != str:
+                data[key] = value
+                continue
+
+            if "{" in value or "[" in value:
```

Comment:
This seems very prone to edge cases – one `[` or `{` in the payload and the data may get surprisingly misinterpreted.